tests/h/services/search_index/_queue_test.py

```diff
@@ -0,0 +1,364 @@
+            annotation_ids[:limit],
+            tag="test",
+            scheduled_at=five_minutes_ago,
+        )
+
+        queue.sync()
+
+        assert caplog.record_tuples == [
+            (
+                Any(),
+                Any(),
+                "Deleting 10 successfully completed jobs from the queue",
+            )
+        ]
+        assert db_session.query(Job).all() == []
+        batch_indexer.index.assert_not_called()
+
+    def test_if_the_annotation_has_a_different_updated_time_in_Elastic_it_indexes_it(
+        self,
+        annotations,
+        annotation_ids,
+        batch_indexer,
+        caplog,
+        es,
+        queue,
+    ):
+        es.conn.search.return_value = {
+            "hits": {
+                "hits": [
+                    {
+                        "_id": annotation.id,
+                        "_source": {
+                            "updated": (
+                                annotation.updated - datetime.timedelta(minutes=5)
+                            ).isoformat(),
+                        },
+                    }
+                    for annotation in annotations[:limit]
+                ],
+            },
+        }
+        queue.add_all(
+            annotation_ids[:limit],
+            tag="test",
+            scheduled_at=five_minutes_ago,
+        )
+
+        queue.sync()
+
+        assert caplog.record_tuples == [
+            (
+                Any(),
+                Any(),
+                "Syncing 10 annotations that are different in Elasticsearch",
+            )
+        ]
+        batch_indexer.index.assert_called_once_with(
+            Any.list.containing(annotation_ids[:limit]).only()
+        )
+
+    def test_if_there_are_multiple_jobs_with_the_same_annotation_id(
+        self,
+        annotation_ids,
+        batch_indexer,
+        caplog,
+        es,
+        queue,
+    ):
+        queue.add_all(
+            [annotation_ids[0], annotation_ids[0], annotation_ids[0]],
+            tag="test",
+            scheduled_at=five_minutes_ago,
+        )
+
+        queue.sync()
+
+        # It only retrieves the annotation from Elasticsearch once.
+        es.conn.search.assert_called_once_with(
+            body=Any.dict.containing(
+                {
+                    "query": Any.dict.containing(
+                        {
+                            "ids": {
+                                "values": [annotation_ids[0]],
+                            },
+                        },
+                    ),
+                },
+            ),
+        )
+
+        # It only syncs the annotation to Elasticsearch once.
+        assert caplog.record_tuples == [
+            (
+                Any(),
+                Any(),
+                "Syncing 1 annotations that are missing from Elasticsearch",
+            )
+        ]
+        batch_indexer.index.assert_called_once_with(
+            Any.list.containing([annotation_ids[0]]).only()
+        )
```

Comment:
```suggestion
        batch_indexer.index.assert_called_once_with([annotation_ids[0]])
```

When there's only one item exactly a plain old list match is probably the ticket.

Reply:
:+1: Done